takari-lifecycle-plugin/src/main/java/io/takari/resources/filtering/FilterResourcesProcessor.java

```diff
@@ -117,19 +117,43 @@ public void encode(String value, Writer writer) {
   // workaround for https://github.com/spullara/mustache.java/issues/92
   // performs full-name map lookup before falling back to dot-notation parsing
   private static class MapReflectionObjectHandler extends ReflectionObjectHandler {
+    private final MissingPropertyAction missingPropertyAction;
+
+    public MapReflectionObjectHandler(final MissingPropertyAction missingPropertyAction) {
+      this.missingPropertyAction = missingPropertyAction;
+    }
+
     @Override
     public Wrapper find(final String name, Object[] scopes) {
+      Wrapper result = null;
       for (final Object scope : scopes) {
         if (scope instanceof Map && ((Map) scope).containsKey(name)) {
-          return new Wrapper() {
+          result = new Wrapper() {
             @Override
             public Object call(Object[] scopes) throws GuardException {
               return ((Map) scope).get(name);
             }
           };
+          break;
+        }
+      }
+      if (result == null) {
+        result = super.find(name, scopes);
+      }
+      if (result instanceof MissingWrapper && missingPropertyAction != MissingPropertyAction.empty) {
+        if (missingPropertyAction == MissingPropertyAction.fail) {
+          throw new MustacheException("Missing property '" + name + "' required for filtering");
```

Comment:
don't throw exception, use `buildContext` to report `input` errors instead. this way the build will report all errors, not just the first one, and it will work properly in eclipse too.

for bonus points, it'd be nice to place the error on the specific line:column of the problem, but I am not sure mustache provides this info.